src/devices/sound/tms5220.cpp

```diff
@@ -815,54 +839,14 @@ int tms5220_device::status_read()
 
 ***********************************************************************************************/
 
-int tms5220_device::ready_read()
+bool tms5220_device::ready_read()
 {
 	LOGMASKED(LOG_PIN_READS, "ready_read: ready pin read, io_ready is %d, fifo count is %d, DDIS(speak external) is %d\n", m_io_ready, m_fifo_count, m_DDIS);
-
-	return ((m_fifo_count < FIFO_SIZE)||(!m_DDIS)) && m_io_ready;
-}
-
-
-/**********************************************************************************************
-
-     tms5220_cycles_to_ready -- returns the number of cycles until ready is asserted
-     NOTE: this function is deprecated and is known to be VERY inaccurate.
-     Use at your own peril!
-
-***********************************************************************************************/
-
-int tms5220_device::cycles_to_ready()
-{
-	int answer;
-
-
-	if (ready_read())
-		answer = 0;
+	// if m_true_timing is NOT set (we're in 'hacky instant write mode'), the m_timer_io_ready doesn't run and will never de-assert m_io_ready if the fifo is full, so we need to explicitly check for fifo full here and return the proper value.
+	if (!m_true_timing)
+		return ((m_fifo_count < FIFO_SIZE)||(!m_DDIS)) && m_io_ready; // SEVERE CAVEAT: this makes the assumption that the ready_read was after wsq was 'virtually asserted', so if the fifo has no room in it ready will always return inactive, even if no write happened (i.e., after a read command when the fifo was exactly filled but no write attempted to overfill it) which is inaccurate to hardware and may cause issues! you have been warned!
```

Comment:
Please wrap comments somewhere around column 72 - it makes them a lot more readable.

Reply:
will do. I'll also make "FIFO" consistently capitalized in comments and logs since its horribly inconsistent.